### ISSUE TITLE
`jsx-no-danger-with-children` should not error if children are empty

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#1398]
### Lint Name

`jsx-no-danger-with-children`

### Code Snippet

```ts
        <template
          ref={tplRef}
          dangerouslySetInnerHTML={{
            __html: `foobar`,
          }}
        >
        </template>
```

### Expected Result

No lint error.

### Actual Result

```
error[jsx-no-danger-with-children]: Using JSX children together with 'dangerouslySetInnerHTML' is invalid
 --> /home/lucacasonato/projects/github.com/denoland/deployng/test.tsx:1:1
  | 
1 | <template
  | ^^^^^^^^^
  | 
2 |   ref={tplRef}
  | ^^^^^^^^^^^^^^
  | 
3 |   dangerouslySetInnerHTML={{
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | 
4 |     __html: `foobar`,
  | ^^^^^^^^^^^^^^^^^^^^^
  | 
5 |   }}
  | ^^^^
  | 
6 | >
  | ^
  | 
7 | </template>;
  | ^^^^^^^^^^^
  = hint: Remove the JSX children

  docs: https://lint.deno.land/rules/jsx-no-danger-with-children


Found 1 problem
Checked 1 file
```

### Version

2.1.7+e475749
